Src/Model/Factory/FormatFactory.php

```diff
@@ -0,0 +1,23 @@
+<?php
+namespace Aot\Model\Factory;
+
+class FormatFactory
+{
+    private $format = ['csv','json'];
+    private $path = 'Aot\Model\Adapter\\';
+
+    /**
+     * create an instance of adapter
+     * @param $format
+     * @return instance
+     * @throws \Exception
+     */
+    public function getFormatInstance($format)
+    {
+        if (in_array($format, $this->format)) {
+            $classPath = $this->path . ucfirst($format);
```

Comment:
Several things here,  good concept! 
However the class may not exist and even if it exists u dont know if it implements `AdapterInterface` or not. You can check `is_a` function in php manual

Reply:
i don't understand !